servicetalk-transport-netty/src/main/java/io/servicetalk/transport/netty/NettyIoExecutors.java

```diff
@@ -91,12 +91,35 @@ public static IoExecutor createIoExecutor(ThreadFactory threadFactory) {
         return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(threadFactory);
     }
 
+    public static IoExecutor createIoExecutor(String threadNamePrefix) {
+        return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(threadNamePrefix);
+    }
+
+    /**
+     * Creates a new {@link IoExecutor} with the specified number of {@code ioThreads}.
+     *
+     * @param ioThreads number of threads.
+     * @param threadNamePrefix the name prefix used for the created {@link Thread}s.
+     * @return The created {@link IoExecutor}
+     */
+    public static IoExecutor createIoExecutor(int ioThreads, String threadNamePrefix) {
+        return io.servicetalk.transport.netty.internal.NettyIoExecutors.createIoExecutor(ioThreads, threadNamePrefix);
+    }
+
     /**
      * Creates a new {@link IoExecutor} with the default number of {@code ioThreads}.
      *
      * @return The created {@link IoExecutor}
      */
     public static IoExecutor createIoExecutor() {
-        return createIoExecutor(new NettyIoThreadFactory(NettyIoExecutor.class.getSimpleName()));
+        return createIoExecutor(newIoThreadFactory());
+    }
+
+    private static NettyIoThreadFactory newIoThreadFactory() {
+        return newIoThreadFactory(NettyIoExecutor.class.getSimpleName());
+    }
+
+    private static NettyIoThreadFactory newIoThreadFactory(String prefix) {
```

Comment:
These two methods are not required because you have a corresponding overloads in `internal.NettyIoExecutors`.

Reply:
The expectation is that not everyone would have the `servicetalk-transport-netty-internal` module imported (or only transitively).

Reply:
If they don't have `servicetalk-transport-netty-internal` module imported, they won't be able to use `new NettyIoThreadFactory(prefix)` the line below.

Reply:
I would expect that `servicetalk-transports-netty` is imported as API and `servicetalk-transports-netty-internal` is imported as implementation or is only imported transitively by one of their other imports.

If both are typically imported as API then this makes me wonder why then why there are two `NettyIoExecutors` utility classes.

Reply:
I think we are talking about different things. You introduced two `private` utilities inside `netty.NettyIoExecutors` class, and new overloads in `netty.internal.NettyIoExecutors`. I'm saying that these new private static methods are unnecessary, because you can just invoke corresponding methods from `netty.internal.NettyIoExecutors`.

Reason why we have two `NettyIoExecutors` is that the internal one has access to `EventLoopGroupIoExecutor` and also exposes API to wrap pre-existing `EventLoop` with `IoExecutor` interface. We don't need to make that API public. As the result, `netty.NettyIoExecutors` is a limited variant of `netty.internal.NettyIoExecutors`, that makes public only what is required for users, but hides what is required for us.

Reply:
OK, got it now. It is as I expected, only `netty.NettyIoExecutors` is public.

Reply:
I will have to create a PR for main to keep it in sync.